brush-parser/src/parser.rs

```diff
@@ -587,12 +587,28 @@ peg::parser! {
                 w:word() {
                     ast::CommandPrefixOrSuffixItem::Word(ast::Word::from(w))
                 }
-            )+ { ast::CommandSuffix(s) }
+            )+ e:input_output_pipe_extension()? {
+                let mut s = s;
+                if let Some(e) = e {
+                    s.push(ast::CommandPrefixOrSuffixItem::IoRedirect(e))
+                }
+                ast::CommandSuffix(s)
+            }
 
         rule redirect_list() -> ast::RedirectList =
-            r:io_redirect()+ { ast::RedirectList(r) } /
+            r:io_redirect()+ e:input_output_pipe_extension()? {
```

Comment:
I don't recall offhand; did you confirm that `|&` is actually allowed anywhere a redirect list can be present (e.g., not just where `|` can be used)?

Reply:
Can you give me some examples what to you mean? Now redirect_list is used after compound commands and function bodies